src/patterns.rs

```diff
@@ -259,56 +259,52 @@ fn rewrite_tuple_pat(pats: &[ptr::P<ast::Pat>],
         pat_vec.insert(pos, dotdot);
     }
 
-    if pat_vec.is_empty() {
-        path_str
```

Comment:
What part of the else case is inefficient and should be avoided? It looks to me like any attempt to keep the early return will need duplication of a significant amount of code contained within the else case due to the possibility of `..`.